### ISSUE TITLE
Can't create reproducible pre-signed URLs

### ISSUE THREAD

**Issue by Petah** [aws/aws-sdk-php#1839]
### Version of AWS SDK for PHP?
3.87.18

### Version of PHP (`php -v`)?
7.2.8

### What issue did you see?
It's not possible to create pre-signed URL that have reproducible signatures which means we can't allow the browser to cache the resource without maintaining some kind of state of generated URLs.

You can hover set `ResponseCacheControl` and `ResponseExpires` options which would seem would allow caching, but since the URL changes every time it doesn't work.

When you generate a presigned URL `X-Amz-Signature` changes every time even if every other parameter is the same. I tried jumping into the source and seeing whats going on. It looks like `SignatureV4` takes an option for `start_time` which isn't able to be passed through the `S3Client` so it uses the current time stamp every time. 

https://github.com/aws/aws-sdk-php/blob/8fa68de81738f851e0aa62fbc0a657f2905aa860/src/Signature/SignatureV4.php#L86
https://github.com/aws/aws-sdk-php/blob/8fa68de81738f851e0aa62fbc0a657f2905aa860/src/S3/S3Client.php#L331

```
        $options = [
            'Bucket' => $bucket,
            'Key' => $key,
            'ResponseCacheControl' => 'public',
        ];
        $command = $client->getCommand('GetObject', $options);
        $url = (string) $client->createPresignedRequest($command, $expiry)->getUri();
```

**Comment by srchase** [comment]
@Petah 

Thanks for opening this issue.

I've marked this as a feature request.  The ```start_time``` could be passed through the ```S3Client``` as you suggest, enabling reproducible signatures such as for your use case.

We'll see where this work can be prioritized.  We'd also be happy to review a PR if this is something you wanted to take on.

**Comment by diehlaws** [comment]
@Petah this feature has been implemented in #1841 and is available in the [latest release](https://github.com/aws/aws-sdk-php/releases/tag/3.105.0) so I'll close out this issue. Please do let us know if you run into any trouble using the `start_time` parameter with the newest version of the AWS SDK for PHP.

**Comment by Petah** [comment]
Thanks, tried this out just now. Works well.
